api/v1/tetragon/tetragon.proto

```diff
@@ -45,71 +50,128 @@ message Container {
+    // Cilium identity labels of the Pod.
     repeated string labels = 3;
+    // Container of the Pod from which the process that triggered the event
+    // originates.
     Container container = 4;
-    // pod_labels field contains all the labels of the pod. Note that the labels field
-    // contains Cilium identity labels, which is a subset of pod labels.
+    // Contains all the labels of the pod. Note that the labels field contains
+    // Cilium identity labels, which is a subset of pod labels.
     map<string, string> pod_labels = 5;
 }
 
 message Capabilities {
+    // Permitted set indicates what capabilities the process can use. This is a
+    // limiting superset for the effective capabilities that the thread may
+    // assume. It is also a limiting superset for the capabilities that may be
+    // added to the inheritable set by a thread without the CAP_SETPCAP in its
+    // effective set.
     repeated CapabilitiesType permitted = 1;
+    // Effective set indicates what capabilities are active in a process. This
+    // is the set used by the kernel to perform permission checks for the
+    // thread.
     repeated CapabilitiesType effective = 2;
+    // Inheritable set indicates which capabilities will be inherited by the
+    // current process when running as a root user.
     repeated CapabilitiesType inheritable = 3;
 }
 
 message Namespace {
+    // Inode number of the namespace.
     uint32 inum = 1;
+    // Indicates if namespace belongs to host.
     bool is_host = 2;
 }
 
 message Namespaces {
+    // Hostname and NIS domain name.
     Namespace uts = 1;
+    // System V IPC, POSIX message queues.
     Namespace ipc = 2;
+    // Mount points.
     Namespace mnt = 3;
+    // Process IDs.
     Namespace pid = 4;
+    // Process IDs for children processes.
     Namespace pid_for_children = 5;
+    // Network devices, stacks, ports, etc.
     Namespace net = 6;
+    // Boot and monotonic clocks.
     Namespace time = 7;
+    // Boot and monotonic clocks for children processes.
     Namespace time_for_children = 8;
+    // Cgroup root directory.
     Namespace cgroup = 9;
+    // User and group IDs.
     Namespace user = 10;
 }
 
 message Process {
     // Exec ID uniquely identifies the process over time across all the nodes in the cluster.
     string exec_id = 1;
+    // Process identifier from host PID namespace.
     google.protobuf.UInt32Value pid = 2;
+    // User identifier associated with the process.
     google.protobuf.UInt32Value uid = 3;
+    // Current working directory of the process.
     string cwd = 4;
+    // Absolute path of the executed binary.
     string binary = 5;
+    // Arguments passed to the binary at execution.
     string arguments = 6;
+    // Information about which syscalls generated events, use of internal
+    // Tetragon buffers, errors and more.
+    // TODO: not sure how to documents all the existing flags without dumping
+    // 60 lines of text
     string flags = 7;
```

Comment:
What do you think?

Reply:
It's too bad this is a string instead of a protobuf enum :/ probably too late to change it sadly.

Reply:
@willfindlay we just made a more of debugging flag, we can add a more reliable field later and get rid of this one if developers or other tools request this... so I guess we are fine ;-) !